cargo-lock/src/bin/cargo-lock/main.rs

```diff
@@ -129,9 +131,57 @@ struct TreeCmd {
     #[options(short = "f", help = "input Cargo.lock file to translate")]
     file: Option<PathBuf>,
 
-    /// Dependencies names to draw a tree for
-    #[options(free, help = "dependency names to draw trees for")]
-    dependencies: Vec<package::Name>,
+    /// Show exact package identities (checksums or specific source versions) when available
+    #[options(
+        short = "x",
+        help = "show exact package identies (checksums or specific source versions) when available"
+    )]
+    exact: bool,
+
+    // Show inverse dependencies rather than forward dependencies
+    #[options(
+        short = "i",
+        help = "show inverse dependencies _on_ a package, rather than forward dependencies _of_ a package"
+    )]
+    inverse: bool,
+
+    /// Dependencies names or hashes to draw a tree for
+    #[options(free, help = "dependency names or hashes to draw trees for")]
+    dependencies: Vec<String>,
+}
+
+fn package_matches_name(pkg: &Package, name: &str) -> bool {
+    pkg.name.as_str().contains(name)
```

Comment:
I'm not sure a substring search is appropriate. For example, `cargo-geiger` depends on `cargo` and this would match both if the string `cargo` is supplied.

Is there a good reason not to require an exact match?

Reply:
Just laziness, and you're right, precision is a better default. I'll change it.